lib/internal/util/comparisons.js

```diff
@@ -293,9 +293,9 @@ function keyCheck(val1, val2, strict, memos, iterationType, aKeys) {
   }
 
   if (aKeys.length === 0 &&
-      (iterationType === kNoIterator ||
-        iterationType === kIsArray && val1.length === 0 ||
-        val1.size === 0)) {
+      ((iterationType === kNoIterator ||
+        iterationType === kIsArray) && (val1.length === 0 ||
+        val1.size === 0))) {
```

Comment:
This seems to be a typo. The condition does not seem to be identical to the original one.

Reply:
Oh, yes, this is different. For example, if `aKeys.length === 0` and `iterationType === kNoIterator` are both `true` but the remaining three conditions are all `false`, the current code evaluates to `true` but the modified code evaluates to `false`. Might be good to add a test for this if possible, since all tests pass with this change. If it's not possible to add a test, it may indicate that this conditional can be simplified.

Reply:
Good catch.

So here I found that always `&&` should be within parenthesis in every mixed operators `&&` and `||`.

Reply:
This is a fast path. If it returns false when it should return true, it will still work. Testing this is almost impossible (only the timing could be tested for plus the coverage).